src/Sema.zig

```diff
@@ -6456,9 +6458,15 @@ fn analyzeCall(
                     break :func_ty ptr_info.pointee_type;
                 }
             },
+            .Optional => {
+                var buf: Type.Payload.ElemType = undefined;
+                if (callee_ty.optionalChild(&buf).zigTypeTag() == .Fn) {
+                    try sema.errNote(block, call_src, msg, "unwrap the optional to get the function", .{});
```

Comment:
I'd write something like this to keep it more in line with other similar errors:
```suggestion
                    try sema.errNote(block, call_src, msg, "cannot call optional function type", .{});
                    try sema.errNote(block, call_src, msg, "consider using '.?', 'orelse', or 'if'", .{});
```